[Utilities] Created utility to get an outline of all module features (#1641)
Azure/ResourceModules: docs/wiki/The library - Module overview.md

This section provides an overview of the library's feature set.

---

### _Navigation_

- [Feature table](#feature-table)

---

# Feature table

> | # | Module | RBAC | Locks | Tags | Diag | EP | Pip-Depl. | # children | # lines |

Reviewer: I'd suggest either to expand column names or better a legend to explain e.g. what EP means

Author: I did have bigger names, but Mate asked me to shorten them. A legend seems to be a good compromise - will add

Author: added

This section provides an overview of the library's feature set.

---

### _Navigation_

- [Feature table](#feature-table)
  - [Legend](#legend)

---

# Feature table

> | # | Module | RBAC | Locks | Tags | Diag | EP | Pip-Depl. | # children | # lines |

Author: ```suggestion
| # | Module | RBAC | Locks | Tags | Diag | EP | PIP | # children | # lines |
```

| 100 | MS.Web<p>sites | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |  | [L1:2] | 248 |
| 101 | MS.Web<p>staticSites | :white_check_mark: | :white_check_mark: | :white_check_mark: |  | :white_check_mark: |  |  | 144 |
| Sum | | 78 | 76 | 86 | 46 | 17 | 2 | 137 | 16534 |

## Legend

| Term | Description |
| - | - |
| `Module` | The name of the module |
| `RBAC` | Whether the module can deploy _Role Assignments_. |
| `Locks` | Whether the module can deploy _Locks_. |
| `Tags` | Whether the module can deploy _Tags_. |
| `Diag` | Whether the module can deploy _Diagnostic Settings_. |

> | `EP` | Whether the module can deploy _Private Endpoints_. |

Reviewer: ```suggestion
| `PE` | Whether the module can deploy _Private Endpoints_. |
```

> | `Pip-Depl.` | Whether the module can deploy a _Public IP_ as a secondary resource. |

Reviewer: ```suggestion
| `PIP` | Whether the module can deploy a _Public IP_ as a secondary resource. |
```

| `# children` | The number of children in the given module. Children (if any) are displayed in format `[L1:5, L2:4, L3:1]`. Each item (separated via ',') shows the level of nesting in the front (e.g. L1) and the number of children in this level (separated by a colon ':'). In the previous example, the module has 5 direct children, 4 of them have direct children themselves and 1 of them has 1 more child. |

> | `# lines` | The number of lines in the given module. |

Reviewer: Are those the number of lines of the deploy.bicep? Then I'd suggest something like
```suggestion
| `# lines` | The number of lines in the module Bicep deployment file. |
```
